src/deploy/functions/deploy.ts

```diff
@@ -29,11 +29,27 @@ async function uploadSourceV1(context: args.Context, region: string): Promise<vo
 async function uploadSourceV2(context: args.Context, region: string): Promise<void> {
   const res = await gcfv2.generateUploadUrl(context.projectId, region);
   const uploadOpts = {
-    file: context.functionsSourceV2!,
-    stream: fs.createReadStream(context.functionsSourceV2!),
+    file: context.source!.functionsSourceV2!,
+    stream: fs.createReadStream(context.source!.functionsSourceV2!),
   };
   await gcs.upload(uploadOpts, res.uploadUrl);
-  context.storage = { ...context.storage, [region]: res.storageSource };
+  context.source!.storage = { ...context.source!.storage, [region]: res.storageSource };
+}
+
+function assertPreconditions(context: args.Context, options: Options, payload: args.Payload): void {
+  const assertExists = function (v: unknown, msg?: string): void {
+    const errMsg = `${msg || "Value unexpectedly empty."}`;
+    if (!v) {
+      throw new FirebaseError(
+        errMsg +
+          "This should never happen. Please file a bug at https://github.com/firebase/firebase-tools"
+      );
+    }
+  };
+  assertExists(context.config, "Functions config unexpectedly empty.");
+  assertExists(context.source, "Functions sources unexpectedly empty.");
+  assertExists(context.source?.functionsSourceV1, "Functions v1 source unexpectedly empty.");
```

Comment:
Won't this be missing if there were no functions targeting v1?

Reply:
Resolved by reverting back to the old check style.